File: 1945 Sum of Digits of String After Convert/1945.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        int ans = 0;

        for (char c: s){
            int num = c - 'a' + 1;
            if (num >= 10) num = num / 10 + num % 10;
            ans += num;
        }

        while (--k){
            int cur = ans;
            ans = 0; 
            while (cur){
                ans += cur % 10;
                cur /= 10;
            }
        }
        
        return ans;
    }
};

File: 1945 Sum of Digits of String After Convert/1945.py
class Solution:
    def getLucky(self, s: str, k: int) -> int:
        ans = 0
        
        for c in s:
            num = ord(c) - ord('a') + 1
            if num >= 10: num = num // 10 + num % 10
            ans += num
        
        while k > 1:
            cur = ans
            ans = 0
            while cur > 0:
                ans += cur % 10
                cur //= 10
            k -= 1

        return ans
